detect_secrets/plugins/common/util.py

```diff
@@ -1,53 +1,149 @@
+import importlib.util
+import inspect
 import os
 from abc import abstractproperty
 from functools import lru_cache
-from importlib import import_module
 
 from detect_secrets.plugins.base import BasePlugin
 from detect_secrets.util import get_root_directory
 
 
+def change_custom_plugin_paths_to_tuple(custom_plugin_paths_function):
```

Comment:
This seems kinda excessive, can we just call `tuple(argument)` wherever we want to try to be safer?

One alternative is to make it so that the `ArgumentParser` returns a `tuple` by implementing a [custom action](https://docs.python.org/3.7/library/argparse.html#action) but honestly I'd be OK with us just not worry about the accidentally mutating the list of paths.

Reply:
The reason for tuplifying is [so that we can `lru_cache` the functions this decorator is used on](https://github.com/Yelp/detect-secrets/pull/255#discussion_r339195196), not b/c I'm afraid the list will get mutated.

I'll re-read the code to see if tuplifying as soon as we parse args (/the baseline) makes things easier. 👍 Convert to X as early as possible seems like always good advice, I'll have to see if there was a reason I wrote it this way. 🤔

Reply:
It does not appear there was a good reason 😁 I'll go ahead and do the custom action, something similar to https://stackoverflow.com/questions/50365835/how-to-make-argparse-argumentparser-return-a-tuple-or-an-np-array-rather-than-a